apps/api/src/domain/models/video/commands/translate-video-name/translate-video-name.command-handler.ts

```diff
@@ -0,0 +1,36 @@
+import { CommandHandler, ICommand } from '@coscrad/commands';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { Video } from '../../../audio-item/entities/video.entity';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { TranslateVideoName } from './translate-video-name.command';
+import { videoNameTranslated } from './video-name-translated.event';
+
+@CommandHandler(TranslateVideoName)
+export class TranslateVideoNameCommandHandler extends BaseUpdateCommandHandler<Video> {
+    protected async fetchRequiredExternalState(_: TranslateVideoName): Promise<InMemorySnapshot> {
+        return Promise.resolve(new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat());
+    }
+
+    protected validateExternalState(
+        _state: InMemorySnapshot,
+        _instance: Video
+    ): InternalError | Valid {
+        return Valid;
+    }
+
+    protected actOnInstance(
+        Video: Video,
+        { text, languageCode }: TranslateVideoName
+    ): ResultOrError<Video> {
+        return Video(text, languageCode);
```

Comment:
You'll want to add the behaviour onto the `Video` class (video.entity.ts) as usual.

```ts
return video.translateName(text,languageCodce)
```

video.entity.ts
```ts
translateName(text: string,languageCode:LanguageCode): ResultOrError(video){
    const nameUpdateResult= ... // use this.name to build a new name

    if(isInternalError(newName)) return nameUpdateResult;

   return this.clone(
    {
        name: newName
    }
)

}
```